src/main/java/de/tum/cit/aet/artemis/exercise/repository/StudentParticipationRepository.java

```diff
@@ -1361,4 +1361,48 @@ SELECT MAX(pr.id)
                 ORDER BY p.student.firstName ASC
             """)
     Page<FeedbackAffectedStudentDTO> findAffectedStudentsByFeedbackId(@Param("exerciseId") long exerciseId, @Param("feedbackIds") List<Long> feedbackIds, Pageable pageable);
+
+    /**
+     * Retrieves the logins of students affected by a specific feedback detail text in a given exercise.
+     *
+     * @param exerciseId The ID of the exercise for which affected students are requested.
+     * @param detailText The feedback detail text to filter by.
+     * @return A list of student logins affected by the given feedback detail text in the specified exercise.
+     */
+    @Query("""
+                SELECT DISTINCT p.student.login
+                FROM ProgrammingExerciseStudentParticipation p
+                LEFT JOIN p.submissions s
+                LEFT JOIN s.results r
+                LEFT JOIN r.feedbacks f
+                WHERE p.exercise.id = :exerciseId
+                  AND f.detailText = :detailText
+                  AND p.testRun = FALSE
+                  AND f.positive = FALSE
```

Comment:
Why do we check for `f.positive = FALSE`? I think `f.detailText = :detailText` cannot be the case if `f.positive = TRUE`.